src/app-layer-ssl.c

```diff
@@ -1025,6 +1043,9 @@ static inline int TLSDecodeHSHelloExtensionSupportedVersions(SSLState *ssl_state
             uint16_t ver = (uint16_t)(input[i] << 8) | input[i + 1];
             if (TLSVersionValid(ver)) {
                 ssl_state->curr_connp->version = ver;
+                if (ssl_state->curr_connp->ja4 != NULL) {
+                    SCJA4SetTLSVersion(ssl_state->curr_connp->ja4, ver);
```

Comment:
We should max here, right ?

Reply:
Well spotted. Now that I think about it, we could also do that in the `JA4` state instead of in the parser. It can keep track of the maximum itself.